designs/ami-selector-reqs.md

```diff
@@ -0,0 +1,244 @@
+      already selects AMIs based on instance compatibility and recency. These requirements act as an additional filter
+      on the AMI's applicability in a given scheduling context. Today, except alias-based terms, these constraints are
+      limited to what can be statically inferred from the AMI itself (e.g., architecture). Therefore, including
+      requirements as a field within an `amiSelectorTerm` is consistent with its purpose: to define the **AND**ed
+      conditions under which an AMI is valid for use.
+
+### Nesting the AMI selector and node requirements separately
+
+This option proposes introducing a nested structure to separate AMI selection from scheduling constraints.
+
+```yaml
+amiSelectorTerms:
+  - selector:
+      ssmParameter: "/my-org/amis/custom-ml-drivers"
+    requirements:
+      - key: kubernetes.io/arch
+        operator: In
+        values: [ 'amd64' ]
+      - key: "karpenter.k8s.aws/instance-gpu-count"
+        operator: "Exists"
+```
+
+* **Pro**: The association between an AMI and its requirements is declarative and self-contained.
+* **Pro**: Clear separation of concerns between AMI selection and scheduling logic.
+* **Con**: Breaking change to the `EC2NodeClass` schema. Migration could be handled via a new API version or an opt-in
+  feature gate.
+
+### SSM-Encoded Requirements
+
+This solution suggests embedding scheduling requirements directly into the SSM Parameter Store metadata associated with
+an AMI.
+
+Given an SSM parameter (e.g. `/my-org/amis/custom-ml-drivers`)
+
+One could then encode scheduling requirements into the SSM parameter as a JSON string, for example:
+
+```
+{"id":"ami-02211404f5aa0b76a","requirements":[{"key":"kubernetes.io/arch","operator":"In","values":["arm64"]},{"key":"karpenter.k8s.aws/instance-gpu-count","operator":"DoesNotExist"},{"key":"karpenter.k8s.aws/instance-accelerator-count","operator":"DoesNotExist"}]}
+```
+
+<details>
+<summary>Click to see formatted JSON...</summary>
+<pre><code>
+{
+    "id": "ami-02211404f5aa0b76a",
+    "requirements": [
+        {
+            "key": "kubernetes.io/arch",
+            "operator": "In",
+            "values": [
+                "arm64"
+            ]
+        },
+        {
+            "key": "karpenter.k8s.aws/instance-gpu-count",
+            "operator": "DoesNotExist"
+        },
+        {
+            "key": "karpenter.k8s.aws/instance-accelerator-count",
+            "operator": "DoesNotExist"
+        }
+    ]
+}
+</code></pre>
+</details>
+
+Karpenter would then resolve the AMI ID and the associated requirements from the SSM parameter, passing the discovered
+requirements on to the resolver, similar to how it does for alias.
+
+* **Pro**: Consistent with the existing `alias` behavior.
+* **Neutral**: Introduces new conventions around SSM parameter format when used as an `amiSelectorTerm`.
+  SSM parameters for well known AMI families (e.g. `bottlerocket`) do not follow this proposed convention.
```

Comment:
If we were to go forward with this approach, I would like to see if we (the Karpenter team at AWS), would be able to align with the teams that publish our officially supported AMIs to support this as well.